src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = []
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+  static removeDead(animal) {
+    Animal.alive = Animal.alive.filter((aliveAnimal) => aliveAnimal !== animal);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(targetAnimal) {
+    if (targetAnimal instanceof Carnivore
+      || (targetAnimal instanceof Herbivore && targetAnimal.hidden)) {
```

Comment:
You can reduce the number of conditions a little
```suggestion
    if (targetAnimal instanceof Carnivore || targetAnimal.hidden)) {
```